client/my-sites/earn/ads/form-settings.tsx

```diff
@@ -69,30 +69,15 @@ const AdsFormSettings = () => {
 	const isWordAds = site?.options?.wordads;
 
 	useEffect( () => {
-		// Set default settings if settings are empty
-		const settingsAreEmptyAndWordadsSettingsExist =
-			isEmpty( settings ) && ! isEmpty( wordadsSettings );
-
-		if ( settingsAreEmptyAndWordadsSettingsExist ) {
-			setSettings( {
-				...defaultSettings(),
-				...wordadsSettings,
-			} );
-		}
-	}, [ settings, wordadsSettings ] );
-
-	useEffect( () => {
-		// siteId has changed, so we reset settings
-		// This code is here to directly mimic code that
-		// was in UNSAFE_componentWillReceiveProps
-		setSettings( {
+		const newSettings = {
 			...defaultSettings(),
 			...wordadsSettings,
-		} );
-
-		// Only run on siteId change
-		// eslint-disable-next-line react-hooks/exhaustive-deps
-	}, [ siteId ] );
```

Comment:
I was wonder if anything would break by not checking siteId here. I don't know exactly why this check was here before I refactored (see https://github.com/Automattic/wp-calypso/pull/81048) but I tried to duplicate the check siteId here without the unsafe lifecylce methods in the old code. 

The only scenario I could think of is if a user tries switching to a separate blog while on this page. But I tested that with this new code, and that seems to work correctly in any case.

Reply:
That's correct. That was the same scenario I tested with and the only one I could think of. I find it okay to remove this code because the updated `useEffect` hook depends on changes to `wordadsSettings` and wordadsSettings value will be updated on changes to `siteId` and that automatically causes the hook to re-run on `siteId` change https://github.com/Automattic/wp-calypso/blob/01d6bc35e8520eec4493a6bad6b0e877d54337ee/client/my-sites/earn/ads/form-settings.tsx#L65